### ISSUE TITLE
Write unit tests

### ISSUE THREAD

**Issue by ionut-arm** [parallaxsecond/rust-tss-esapi#10]
There are essentially no real unit tests that reliably cover the functionality we expose. Should be fixed before we can upstream.

**Comment by hug-dev** [comment]
And integration tests as well 😋

**Comment by hug-dev** [comment]
As we discovered that it was needed to re-set the session attribute for every command, the following test might be nice to have:
```rust
    #[test]
    fn transient_test() {
        let mut ctx = TransientObjectContext::new(Tcti::Mssim, 2048, 32, &[]).unwrap();
        let key = match ctx.create_rsa_signing_key(2048, 32) {
            Ok((key, _auth)) => key,
            Err(e) => {
                println!("{}\n{:?}", e, e);
                panic!();
            }
        };
        let signature = Signature {
            scheme: AsymSchemeUnion::RSASSA(TPM2_ALG_SHA256),
            signature: vec![1, 2, 3, 5],
        };
        if let Err(e) = ctx.verify_signature(key, &HASH, signature) {
            println!("{}\n{:?}", e, e);
        }
        if let Err(e) = ctx.context.get_random(1) {
            println!("{}\n{:?}", e, e);
        }
    }
```

**Comment by ionut-arm** [comment]
Covered for now as per #26 and #25
